pyscript.core/docs/README.md

```diff
@@ -0,0 +1,341 @@
+| target    | `<script type="pyodide" target="outcome">`    | Describe as *id* or *CSS* selector the default *target* to use as `document.currentScript.target` field. Please read the [Terminology](#terminology) **target** dedicated details to know more. |
+| type      | `<script type="micropython">`                 | Define the *interpreter* to use with this script. Please read the [Terminology](#terminology) **interpreter** dedicated details to know more. |
+| version   | `<script type="pyodide" version="0.23.2">`    | Allow the usage of a specific version where, if numeric, must be available through the project *CDN* used by *core* but if specified as fully qualified *URL*, allows usage of any interpreter's version: `<script type="pyodide" version="http://localhost:8080/pyodide.local.mjs">` |
+
+
+### Script Features
+
+These are all special, *script* related features, offered by *@pyscript/core* out of the box.
+
+<details>
+  <summary><strong>document.currentScript</strong></summary>
+  <div>
+
+No matter the interpreter of choice, if there is any way to reach the `document` from such interpreter, its `currentScript` will point at the exact/very-same script that is currently executing the code, even if its `async` attribute is used, mimicking what the standard [document.currentScript](https://developer.mozilla.org/en-US/docs/Web/API/Document/currentScript) offers already, and in an unobtrusive way for the rest of the page, as this property only exists for *synchronous* and blocking scripts that are running, hence never interfering with this *core* logic or vice-versa.
+
+```html
+<script type="micropython" id="my-target">
+    from js import document
+
+    # explicitly grab the current script as target
+    my_target = document.getElementById('my-target')
+
+    # verify it is the exact same node with same id
+    print(document.currentScript.id == my_target.id)
+</script>
+```
+
+Not only this is helpful to crawl the surrounding *DOM* or *HTML*, every script will also have a `target` property that will point either to the element reachable through the `target` attribute, or it lazily creates once a companion element that will be appended right after the currently executing *script*.
+
+Please read the [Terminology](#terminology) **target** dedicated details to know more.
+
+  </div>
+</details>
+
+<details>
+  <summary><strong>XWorker</strong></summary>
+  <div>
+
+With or without access to the `document`, every (*non experimental*) interpreter will have defined, at the global level, a reference to the `XWorker` "_class_" (it's just a *function*!), which goal is to enable off-loading heavy operations on a worker, without blocking the main / UI thread (the current page) and allowing such worker to even reach the `document` or anything else available on the very same main / UI thread.
```

Comment:
I think this is the first reference to experimental vs non-experimental interpreters. Do we intend to release any interpreters that are still experimental? If not, I don't think we need to mention them here.

Reply:
Mmmm... this is the Tech Preview docs. I assume we'll re-work them and change as we work on the next task: "Moving `next` as new core and make the right architecture and high level API changes". The XWorker interface, for instance might change... or we may decide to move the experimental types out of core and make them available as plugins out of core? 

Just want to make sure we are all in agreement that this is not the final/set in stone, architecture/API.

Reply:
I'd also rewrite this in a way to put more emphasis in the `XWorker` API itself rather than the interpreter. Maybe something like:

"
The goal of the`XWorker` API is to enable off-loading heavy operations to a worker, without blocking the main / UI thread (the current page), while also allowing such worker to even reach the `document` or anything else available on the very same main / UI thread.

Note that every (*non experimental*) interpreter will have defined, at the global level, a reference to the `XWorker`, with or without access to the `document`.
"